homeassistant/components/axis/device.py

```diff
@@ -95,32 +97,48 @@ def serial(self):
             self.hass.async_create_task(
                 self.hass.config_entries.async_forward_entry_setup(
                     self.config_entry, 'binary_sensor'))
+
+            self.api.stream.connection_status_callback = \
+                self.async_connection_status_callback
+            self.api.enable_events(event_callback=self.async_event_callback)
             self.api.start()
 
         return True
 
     @callback
-    def async_signal_callback(self, action, event):
+    def async_connection_status_callback(self, status):
+        """Handle signals of gateway connection status."""
+        from axis.streammanager import SIGNAL_PLAYING
+        if self.available != status == SIGNAL_PLAYING:
+            self.available = status == SIGNAL_PLAYING
+            async_dispatcher_send(self.hass, AXIS_REACHABLE, 0)
+
+    @callback
+    def async_event_callback(self, action, event):
         """Call to configure events when initialized on event stream."""
         if action == 'add':
-            async_dispatcher_send(self.hass, 'axis_add_sensor', event)
+            async_dispatcher_send(self.hass, AXIS_ADD_SENSOR, event)
+
+    @callback
+    def retry(self):
```

Comment:
Where is this used?

Reply:
Something I wasn't ready to migrate to. Gonna remove all smart behaviour from libs. Gonna remove it for now